src/components/menu/menu.component.js

```diff
@@ -1,8 +1,80 @@
-import React from 'react';
+import React, { useRef, useCallback, useState } from 'react';
 import PropTypes from 'prop-types';
 import { StyledMenuWrapper, StyledMenuItemsWrapper, StyledMenuItem } from './menu.style';
+import Events from '../../utils/helpers/events';
 
 const Menu = ({ menuType = 'light', children }) => {
+  const menuItemsRefs = useRef(React.Children.map(children, child => child.ref || React.createRef()));
+  const actualFocusedItemIndex = useRef();
+  const [isOpenSubmenu, setIsOpenSubmenu] = useState(null);
+  const setFocusToElement = useCallback((event, index) => {
+    if (event) {
+      event.preventDefault();
+    }
+
+    actualFocusedItemIndex.current = index;
+    const isFocusedOnLastMenuItem = index === menuItemsRefs.current.length;
+    const isFocusedOnFirstMenuItem = index === -1;
+    if (isFocusedOnLastMenuItem) {
+      actualFocusedItemIndex.current = 0;
+    } else if (isFocusedOnFirstMenuItem) {
+      actualFocusedItemIndex.current = menuItemsRefs.current.length - 1;
+    }
+
+    menuItemsRefs.current[actualFocusedItemIndex.current].current.focus();
+  }, []);
+
+  const handleKeyDown = useCallback((event, index) => {
+    if (Events.isRightKey(event)) {
+      event.preventDefault();
+      setFocusToElement(event, index + 1);
+    }
+
+    if (Events.isLeftKey(event)) {
+      event.preventDefault();
+      setFocusToElement(event, index - 1);
+    }
+
+    if (!isOpenSubmenu) {
+      if (Events.isHomeKey(event)) {
+        event.preventDefault();
+        menuItemsRefs.current[0].current.focus();
+      } else if (Events.isEndKey(event)) {
+        event.preventDefault();
+        menuItemsRefs.current[menuItemsRefs.current.length - 1].current.focus();
+      } else if (Events.isAlphabetKey(event)) {
+        // A-Za-z: focus the next item on the list that starts with the pressed key
+        // selection should wrap to the start of the list
+        event.stopPropagation();
+        let firstMatch;
+        let nextMatch;
+        const hasSubmenu = (element) => {
```

Comment:
`hasSubmenu` - it should be renamed to something more descriptive like `getMenuText` or `getMenuLabel`

Reply:
You are right, I think that `getMenuText` is fine